stock_reorder_forecast/models/product_product.py

```diff
@@ -166,33 +166,22 @@ def calc_purchase_date(self):
                 TP.product_id AS id,
                 MAX(TP.stock_period_min) AS stock_period_min,
                 COALESCE(
-                    SUM(SOL.product_uom_qty), 0) /
-                    MAX(CASE WHEN TP.turnover_period < TP.prod_age
-                        THEN TP.turnover_period
-                    WHEN TP.prod_age > TP.turnover_period THEN TP.prod_age
-                    ELSE 1 END)
-                        AS turnover_average,
-                    MAX(TP.stock_period_max) AS stock_period_max
+                    SUM(SOL.product_uom_qty), 0
+                ) /
+                GREATEST(
+                    LEAST(MAX(TP.turnover_period), MAX(TP.prod_age)), 1
```

Comment:
`max` is an aggregate function, when you put this as outermost function, you get the maximum of your expression over all records involved, but what you want here is the maximum of a couple of values